File: ViewModel/Types/SetResultInput.vb
Imports POFF.Kicker.Model

Namespace Types

    Public Class SetResultInput

        Public Sub New()
            HomeValue = Nothing
            GuestValue = Nothing
        End Sub

        Public Sub New(setResult As SetResult)
            If setResult Is Nothing Then Throw New ArgumentNullException("setResult")

            HomeValue = setResult.Home
            GuestValue = setResult.Guest
        End Sub

        Private HomeValue As Integer?
        Public Property Home As Integer?
            Get
                Return HomeValue
            End Get
            Set(value As Integer?)
                HomeValue = value
            End Set
        End Property

        Private GuestValue As Integer?
        Public Property Guest As Integer?
            Get
                Return GuestValue
            End Get
            Set(value As Integer?)
                GuestValue = value
            End Set
        End Property

    End Class

End Namespace

File: POFF.Kicker/Domain/SetResultInput.cs
using System;

namespace POFF.Kicker.Domain;

public class SetResultInput
{
    public SetResultInput()
    {
        Home = default;
        Guest = default;
    }

    public SetResultInput(SetResult setResult)
    {
        if (setResult is null)
            throw new ArgumentNullException("setResult");

        Home = setResult.Home;
        Guest = setResult.Guest;
    }

    public int? Home { get; set; }

    public int? Guest { get; set; }
}